File: reporters/src/main/java/com/facebook/battery/reporter/devicebattery/DeviceBatteryMetricsReporter.java
/**
 * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
 *
 * <p>This source code is licensed under the BSD-style license found in the LICENSE file in the root
 * directory of this source tree. An additional grant of patent rights can be found in the PATENTS
 * file in the same directory.
 */
package com.facebook.battery.reporter.devicebattery;

import com.facebook.battery.metrics.devicebattery.DeviceBatteryMetrics;
import com.facebook.battery.reporter.api.SystemMetricsReporter;

public class DeviceBatteryMetricsReporter implements SystemMetricsReporter<DeviceBatteryMetrics> {

  public static final String BATTERY_PCT = "battery_pct";
  public static final String BATTERY_REALTIME_MS = "battery_realtime_ms";
  public static final String CHARGING_REALTIME_MS = "charging_realtime_ms";

  @Override
  public void reportTo(DeviceBatteryMetrics metrics, SystemMetricsReporter.Event event) {
    event.add(BATTERY_PCT, metrics.batteryLevelPct);
    event.add(BATTERY_REALTIME_MS, metrics.batteryRealtimeMs);
    event.add(CHARGING_REALTIME_MS, metrics.chargingRealtimeMs);
  }
}

File: reporters/src/main/java/com/facebook/battery/reporter/devicebattery/DeviceBatteryMetricsReporter.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.battery.reporter.devicebattery

import com.facebook.battery.metrics.devicebattery.DeviceBatteryMetrics
import com.facebook.battery.reporter.core.SystemMetricsReporter

class DeviceBatteryMetricsReporter : SystemMetricsReporter<DeviceBatteryMetrics> {

  override fun reportTo(metrics: DeviceBatteryMetrics, event: SystemMetricsReporter.Event) {
    event.add(BATTERY_PCT, metrics.batteryLevelPct.toDouble())
    event.add(BATTERY_REALTIME_MS, metrics.batteryRealtimeMs)
    event.add(CHARGING_REALTIME_MS, metrics.chargingRealtimeMs)
  }

  companion object {
    const val BATTERY_PCT: String = "battery_pct"
    const val BATTERY_REALTIME_MS: String = "battery_realtime_ms"
    const val CHARGING_REALTIME_MS: String = "charging_realtime_ms"
  }
}
